hail/src/main/scala/is/hail/types/physical/PCanonicalNDArray.scala

```diff
@@ -84,35 +84,31 @@ final case class PCanonicalNDArray(elementType: PType, nDims: Int, required: Boo
     ))
   }
 
-  def makeColumnMajorStridesBuilder(sourceShapeArray: IndexedSeq[Value[Long]], mb: EmitMethodBuilder[_]): StagedRegionValueBuilder => Code[Unit] = { srvb =>
-    val runningProduct = mb.newLocal[Long]()
-    val tempShapeStorage = mb.newLocal[Long]()
+  def makeColumnMajorStridesBuilder(sourceShapeArray: IndexedSeq[Value[Long]], cb: EmitCodeBuilder): StagedRegionValueBuilder => Code[Unit] = { srvb =>
```

Comment:
This function takes a codebuilder but returns a function to Code[Unit]. I think this is invalid. It might work because all you're doing with the cb is making a local variable (and cb.emb is the same MB as srvb.mb probably), but we shouldn't open ourselves up to misuse like this.